web-client/src/app/state/session-xrpl.service.ts

```diff
@@ -423,39 +467,54 @@ export class SessionXrplService {
 
   protected async prepareUnsignedTransaction(
     receiverId: string,
-    amount: xrpl.Payment['Amount']
+    amount: xrpl.Payment['Amount'],
+    sender?: string
   ): Promise<xrpl.Payment> {
     const { wallet } = this.sessionQuery.assumeActiveSession();
+    const senderId = sender
+      ? sender
+      : this.sessionQuery.assumeActiveSession().wallet.xrpl_account.address_base58;
```

Comment:
Should use `wallet` variable in line 473 and simplify this to `wallet.xrpl_account.address_base58`